app/controllers/concerns/report_actions.rb

```diff
@@ -82,10 +82,12 @@ def generate_module_contents_json(my_module)
                                           .singularize}_element.html.erb",
             contents.parse_locals([report_el])
           )
-          if contents.element == :step
-            el[:children] = generate_step_contents_json(report_el)
-          elsif contents.element == :result
-            el[:children] = generate_result_contents_json(report_el)
+          contents.locals.each do |locals|
+            if locals == :step
+              el[:children] = generate_step_contents_json(report_el)
+            elsif locals == :result
+              el[:children] = generate_result_contents_json(report_el)
+            end
```

Comment:
I think it's safe to just do `contents.locals.first`, and in that case you don't need a loop.